apps/src/code-studio/progressRedux.js

```diff
@@ -84,18 +85,51 @@ const initialState = {
 /**
  * Thunks
  */
+
+// The user has navigated to a new level in the current lesson,
+// so we should update the browser and also set this as the new
+// current level.
 export function navigateToLevelId(levelId) {
   return (dispatch, getState) => {
     const state = getState().progress;
-    const newLevel = state.lessons[0].levels.find(level =>
-      level.ids.find(id => id === levelId)
-    );
+    const newLevel = state.lessons
+      .find(lesson => lesson.id === state.currentLessonId)
+      .levels.find(level => level.ids.find(id => id === levelId));
 
     updateBrowserForLevelNavigation(state, newLevel.url, levelId);
     dispatch(setCurrentLevelId(levelId));
   };
 }
 
+// The user has successfully completed the level and the page
+// will not be reloading.
+export function sendSuccessReport(appType) {
+  return (dispatch, getState) => {
+    const state = getState().progress;
+    const levelId = state.currentLevelId;
+    const currentLevel = state.lessons
+      .find(lesson => lesson.id === state.currentLessonId)
+      .levels.find(level => level.ids.find(id => id === levelId));
+    const scriptLevelId = currentLevel.id;
+
+    const data = {
+      app: appType,
+      result: true,
+      testResult: 100,
+    };
+
+    return $.ajax({
+      type: 'POST',
+      url: `/milestone/0/${scriptLevelId}/${levelId}`,
```

Comment:
What's the `0` here?

Reply:
In the [route](https://github.com/code-dot-org/code-dot-org/blob/49438f119d0551b5aa4c23b0f4860b395696a463/dashboard/config/routes.rb#L524) it's the user ID, though in other labs we also seem to let the server determine this and simply pass 0.  I'll make this a constant with a comment.